client/me/privacy/dpa.jsx

```diff
@@ -0,0 +1,133 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { useMutation } from 'react-query';
+import { useTranslate } from 'i18n-calypso';
+
+/**
+ * Internal dependencies
+ */
+import { Button, Card } from '@automattic/components';
+import SectionHeader from 'calypso/components/section-header';
+import { useDispatch } from 'react-redux';
+import { errorNotice, successNotice } from 'calypso/state/notices/actions';
+import wp from 'calypso/lib/wp';
+
+const useSuccessNotice = ( isSuccess ) => {
+	const showNotice = React.useRef();
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	React.useEffect( () => {
+		showNotice.current = () => {
+			dispatch(
+				successNotice(
+					translate( 'Request successful! We are sending you our DPA via email', {
+						comment:
+							'A Data Processing Addendum (DPA) is a document to assure customers, vendors, and partners that their data handling complies with the law.',
+					} ),
+					{ id: 'request-dpa-notice' }
+				)
+			);
+		};
+	}, [ dispatch, translate ] );
+
+	React.useEffect( () => {
+		isSuccess && showNotice.current();
+	}, [ isSuccess ] );
+};
+
+const useErrorNotice = ( isError, error ) => {
+	const showNotice = React.useRef();
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	React.useEffect( () => {
+		showNotice.current = () => {
+			dispatch(
+				errorNotice(
+					error.error === 'too_many_requests'
+						? error.message
+						: translate( 'There was an error requesting a DPA', {
+								comment:
+									'A Data Processing Addendum (DPA) is a document to assure customers, vendors, and partners that their data handling complies with the law.',
+						  } ),
+					{ id: 'request-dpa-notice' }
+				)
+			);
+		};
+	}, [ dispatch, translate, error ] );
+
+	React.useEffect( () => {
+		isError && showNotice.current();
+	}, [ isError ] );
+};
+
+const useRequestDpaMutation = () =>
```

Comment:
this seemed so small I opted for keeping it here but am open to moving it to `client/data`.